README.md

```diff
@@ -1666,6 +1666,106 @@ With all of our builtin logic integrated into the codebase, we can implement any
 
 [Next sections: Implementing each builtin runner]
 
-#### Hints
 
-TODO
+### Implementing each builtin runner 
+
+
+#### Bitwise builtin
+
+This builtin provides a way to work with the basic bit operations `and`, `or` and `xor`. It implements the basic builtin interface methods:
+
+```go
+func (b *BitwiseBuiltinRunner) Base() memory.Relocatable {
+	return r.base
+}
+
+func (b *BitwiseBuiltinRunner) Name() string {
+	return "range_check"
+}
+```
+
+For the `InitializeSegments` we just add a segment to the memory and store in the base attribute, the first adress of the segment. 
+
+```go
+func (b *BitwiseBuiltinRunner) InitializeSegments(segments *memory.MemorySegmentManager) {
+	r.base = segments.AddSegment()
+}
+```
+
+we also have `InitialStack` method that returns a stack the base address appended
+
+```go 
+func (b *BitwiseBuiltinRunner) InitialStack() []memory.MaybeRelocatable {
+	if b.included {
+		return []memory.MaybeRelocatable{*memory.NewMaybeRelocatableRelocatable(b.base)}
+	} else {
+		return []memory.MaybeRelocatable{}
+	}
+}
+```
+
+The method `DeducedMemoryCell` fetches from memory the operands and  performes the operations. 
+- If the index is lesser than two then the methods just return nil
+- After the operands are fetch it is checked that both are felts because bitwise can not be perform on relocatables or nil values.
```

Comment:
```suggestion
- After the operands are fetched it checks that both of them are felts because bitwise can not be performed on relocatable or nil values.
```